scripts/generate_catalog/generate_catalog.go

```diff
@@ -0,0 +1,321 @@
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "%v\n", err)
+			os.Exit(1)
+		}
+
+		err = copyExamples(b, funcName, funcDest, versionDest)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "%v\n", err)
+			os.Exit(1)
+		}
+
+		err = copyReadme(b, funcName, relativeFuncPath, versionDest)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "%v\n", err)
+			os.Exit(1)
+		}
+
+		f := functions[funcName]
+		f.FunctionName = funcName
+		if f.VersionToExamples == nil {
+			f.VersionToExamples = make(map[string]map[string]example)
+		}
+		metadataPath := strings.TrimSpace(fmt.Sprintf("%v:%v", b, filepath.Join(relativeFuncPath, "metadata.yaml")))
+		f, err = parseMetadata(f, metadataPath, minorVersion, versionDest)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "%v\n", err)
+			os.Exit(1)
+		}
+		functions[funcName] = f
+	}
+
+	flattenedFunctions := make([]function, 0)
+	for _, f := range functions {
+		flattenedFunctions = append(flattenedFunctions, f)
+	}
+	sort.Slice(flattenedFunctions, func(i int, j int) bool {
+		return flattenedFunctions[i].FunctionName < flattenedFunctions[j].FunctionName
+	})
+	return flattenedFunctions
+}
+
+func copyExamples(b string, funcName string, funcDest string, versionDest string) error {
+	exampleSource := fmt.Sprintf("examples/%v", funcName)
+
+	// Prepare destination for versioned examples.
+	err := os.MkdirAll(funcDest, 0744)
+	if err != nil {
+		return err
+	}
+
+	// Copy examples for the function's version to a temporary directory.
+	tempDir, err := os.MkdirTemp("", "examples")
+	if err != nil {
+		return err
+	}
+	cmd := exec.Command("git", fmt.Sprintf("--work-tree=%v", tempDir), "checkout", b, "--", exampleSource)
+	err = cmd.Run()
+	if err != nil {
+		return err
+	}
+
+	// Move example content to the site's example directory.
+	err = os.Rename(filepath.Join(tempDir, "examples", funcName), versionDest)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func copyReadme(b string, funcName string, relativeFuncPath string, versionDest string) error {
+	// Copy README for the function's version to the example directory.
+	tempDir, err := os.MkdirTemp("", "functions")
+	if err != nil {
+		return err
+	}
+	cmd := exec.Command("git", fmt.Sprintf("--work-tree=%v", tempDir), "checkout", b, "--", filepath.Join(relativeFuncPath, "README.md"))
+	err = cmd.Run()
+	if err != nil {
+		return err
+	}
+
+	// Find the README in the example directory.
+	m, err := filepath.Glob(filepath.Join(tempDir, "functions", "*", funcName, "README.md"))
+	if err != nil {
+		return err
+	}
+
+	// Move the README to the destination directory.
+	err = os.Rename(m[0], filepath.Join(versionDest, "README.md"))
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func parseMetadata(f function, metadataPath string, version string, versionDest string) (function, error) {
+	var buf bytes.Buffer
+	// Get the content of metadata.yaml from the appropriate release branch.
+	cmd := exec.Command("git", "show", metadataPath)
```

Comment:
Why we use `git show` to get the content of a file? Is using `cat` more straight forward?

Reply:
Switched to `git cat-file` to retrieve the file in a different branch.